rfcs/0000-dsl-driven-rda-code-generation.md

```diff
@@ -0,0 +1,681 @@
+          sqlType: timestamp with time zone
+    transformations:
+      - from: PARENT
+        to: idrClmHdIcn
+      - from: INDEX
+        to: priority
+      - from: idrDiagIcdType
+        transformer: MessageEnum
+        transformerOptions:
+          enumClass: gov.cms.mpsm.rda.v1.mcs.McsDiagnosisIcdType
+          unrecognizedNameSuffix: EnumUnrecognized
+      - from: idrDiagCode
+      - from: NOW
+        to: lastUpdated
+````
+
+*Note: The full DSL file from the POC can be found here: [POC mapping.yaml](https://github.com/CMSgov/beneficiary-fhir-data/blob/brianburton/dcgeo-186-entities-dsl/apps/bfd-model/bfd-model-rda/mapping.yaml)*
+
+This example illustrates some of the advantages of using a declarative file:
+
+- Standard conventions can be supported as defaults.
+For example:
+  - The `to` only needs to be defined if it differs from the `from`.  Generally columns are named directly based on the RDA API field but not always.
+  - A `javaType` only needs to be defined if it differs from the default for the column's data type.
+  - A `transformer` only needs to be defined if it differs from the default for the column's data type.
+- Transformers can be easily created and added to the plugin.  Each has a name used to reference it in the YAML.
+- The plugin follows a simple set of rules to choose a default transformation if no `transformer`  is provided in the YAML.
+- A single field can have multiple transformations.  For example the MBI field can be copied directly to a column and also used to store a hashed value in another column.
+- Transformers can have their own specific options if their behavior is modifiable from default settings.
+- A few structural transforms can be specified using a virtual `from` that triggers the transform.  Essentially these are for fields that are known at runtime but not taken directly from the messages (like array indexes, the current timestamp, parent primary key column value, etc).
+
+Since the RDA API data is used in different modules within the BFD code base the plugin defines goals specific to each type of code that it generates:
+
+- The `entities` goal generates Hibernate database entity classes with all appropriate annotations, getters, setters, equals/hashCode, and entity relationship mappings.
+- The `transformers` goal generates data transformation/validation classes to copy data from RDA API protbuf messages into database entity classes.
+- The `migration` goal generates SQL DDL for each table that can be used as the basis for creating Flyway migration files.
+- The `random-data` goal generates random data generation classes to create randomized data of appropriate size and type for each object/field in the RDA API messages.
+- The `synthea-bridge` goal generates data transformation classes to copy data from Synthea RIF data files into protobuf messages.
+
+For an idea of the code savings consider the difference in complexity between that YAML example and these hand written classes:
```

Comment:
I would consider adding a new subsection here and possibly expand a little bit on code savings.  The links are fine, but maybe you could summarize by comparing the LOC of the DSL vs the 3 files below?